tfhe/src/high_level_api/compact_list.rs

```diff
@@ -366,27 +370,27 @@ pub struct CompactCiphertextListExpander {
     tag: Tag,
 }
 
-impl CompactCiphertextListExpander {
-    pub fn len(&self) -> usize {
+impl CiphertextList for CompactCiphertextListExpander {
+    fn len(&self) -> usize {
         self.inner.len()
     }
 
-    pub fn is_empty(&self) -> bool {
+    fn is_empty(&self) -> bool {
         self.len() == 0
     }
 
-    pub fn get_kind_of(&self, index: usize) -> Option<crate::FheTypes> {
+    fn get_kind_of(&self, index: usize) -> Option<crate::FheTypes> {
         self.inner.get_kind_of(index).and_then(|data_kind| {
             crate::FheTypes::from_data_kind(data_kind, self.inner.message_modulus())
         })
     }
 
-    pub fn get<T>(&self, index: usize) -> Option<crate::Result<T>>
+    fn get<T>(&self, index: usize) -> crate::Result<Option<T>>
```

Comment:
can't we collapse that into a single Result ?

Maybe not for this release as it would change a few callsites, but a bit later, dunno

Reply:
I think @mayeul-zama also had this in mind.

Personally I think they have a slightly different semantic, and having a "None" meaning "not found" makes it in line with the stdlib. But I don't have a strong opinion about that, and I agree that it can simplify the handling for the caller.